hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -2486,6 +2486,23 @@ public Object loadByUniqueKey(
 		return getAppropriateUniqueKeyLoader( propertyName, session ).loadByUniqueKey( session, uniqueKey );
 	}
 
+	public Object loadByUniqueKey(
```

Comment:
This is only used from natural-id loading and at the moment limits the usefulness to single attribute natural-ids.

IMO this should be changed to something like

```
public Object loadByNaturalId(Object key, ...) {
    ...
}
```

Then this can be used for both single- and multi-attribute natural-ids and clean up the event listener

Reply:
Yup, if we're going to actually do this, I plan to generalize it to multiple `@NaturalId`s. However, there is a small difference between the two cases that I need to account for: `createUniqueKeyLoader()` has these lines:

```java
		if ( uniqueKeyType.isEntityType() ) {
			String className = ( (EntityType) uniqueKeyType ).getAssociatedEntityName();
			uniqueKeyType = getFactory().getMetamodel().entityPersister( className ).getIdentifierType();
		}
```

They're wrong for the case of `@NaturalId` loading.

Not quite sure what to do about this difference other than having two different methods.

Reply:
You don't necessarily have to "go through" `createUniqueKeyLoader()` specifically.  In fact, not sure I see a benefit to doing that.  

You may run into similar problems with EntityLoader itself though.